packages/vx-tooltip/src/enhancers/withTooltip.tsx

```diff
@@ -17,6 +17,10 @@ type WithTooltipState<TooltipData> = Pick<
 type ShowTooltipArgs<TooltipData> = Omit<WithTooltipState<TooltipData>, 'tooltipOpen'>;
 type UpdateTooltipArgs<TooltipData> = WithTooltipState<TooltipData>;
 type WithTooltipContainerProps = { style: React.CSSProperties };
+type RenderTooltipContainer = (
+  children: ReactElement,
```

Comment:
Agreed on `ReactElement` -> `ReactNode` 👍

As far as making the `children` type more precise, I'm not sure if it's possible.  In the render method:

```
    render() {
      return renderContainer(
        <BaseComponent
          updateTooltip={this.updateTooltip}
          showTooltip={this.showTooltip}
          hideTooltip={this.hideTooltip}
          {...this.state}
          {...this.props}
        />,
        containerProps,
      );
    }
```

The `<BaseComponent ... />` passed in to `renderContainer` is an element, not a component, so `children: typeof BaseComponent` would expect a component and wouldn't compile.

`ReactElement` does accept generic type args which I thought might be able to constrain the types a bit:

```
type EnhancedComponentProps<BaseComponentProps, TooltipData> = BaseComponentProps &
  WithTooltipProvidedProps<TooltipData>;
type RenderTooltipContainer<
  BaseComponentType extends React.ComponentType<
    EnhancedComponentProps<BaseComponentProps, TooltipData>
  >,
  BaseComponentProps,
  TooltipData
> = (
  children: ReactElement<
    EnhancedComponentProps<BaseComponentProps, TooltipData>,
    BaseComponentType
  >,
  containerProps?: WithTooltipContainerProps,
) => ReactNode;

export default function withTooltip<BaseComponentProps = {}, TooltipData = {}>(
  BaseComponent: React.ComponentType<EnhancedComponentProps<BaseComponentProps, TooltipData>>,
  containerProps: WithTooltipContainerProps = {
    style: {
      position: 'relative',
      width: 'inherit',
      height: 'inherit',
    } as const,
  },
  renderContainer: RenderTooltipContainer<typeof BaseComponent, BaseComponentProps, TooltipData> = (
    children,
    props,
  ) => <div {...props}>{children}</div>,
) 
```

This is a bit cumbersome but compiles fine, and seems like it should constrain the type of `children` based on component type and props:

```
children: ReactElement<
  EnhancedComponentProps<BaseComponentProps, TooltipData>,
  BaseComponentType
>,
```

Unfortunately, `<BaseComponent ... />` in the render method, or anything else we put there (e.g., `<div />`), gets typed as `JSX.Element`, which is defined as:

```
 interface Element extends React.ReactElement<any, any> { }
```

So any element can be passed to `renderContainer` despite the type constraint here.  For example:

```
    render() {
      return renderContainer(
        <div />,
        containerProps,
      );
    }
```

Passes the type check fine, since `<div />` again gets typed as `JSX.Element`.

This might be possible once this TS issue is resolved: https://github.com/microsoft/TypeScript/issues/21699.  Aside from that I can't think of any other good approaches here.

Ultimately it shouldn't affect any consumers of the `withTooltip` API -- it just affects the internal `render` method -- so it's probably not too impactful.  

That turned into a bit of a wall of text 😅.  Let me know if you have any other thoughts/suggestions on it!